Using type alias in generic throws compilation error

### Describe the bug

Using type aliases on a generic LinkedList type throws a compilation error

### Expected Behavior

LinkedList should work with type aliases

### Current Behavior

```v
==================
   ^~~~~~~~~~~~~~~
/tmp/v_501/linkedlist.10909367588917112498.tmp.c:12536:15: error: implicit declaration of function 'Array_int_str' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        string _t1 = Array_int_str(datatypes__LinkedList_T_main__Distance_array_T_main__Distance(list));
                     ^
/tmp/v_501/linkedlist.10909367588917112498.tmp.c:12536:15: note: did you mean 'Array_string_str'?
/tmp/v_501/linkedlist.10909367588917112498.tmp.c:6467:8: note: 'Array_string_str' declared here
string Array_string_str(Array_string a) {
       ^
/tmp/v_501/linkedlist.10909367588917112498.tmp.c:12536:9: error: initializing 'string' (aka 'struct string') with an expression of incompatible type 'int'
        string _t1 = Array_int_str(datatypes__LinkedList_T_main__Distance_array_T_main__Distance(list));
               ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning and 2 errors generated.
...
==================
```

### Reproduction Steps
```v
module main

import datatypes

type Distance = int

struct Foo {
	field datatypes.LinkedList[Distance]
}

fn main() {
}
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

0.3.3

### Environment details (OS name and version, etc.)

macOS

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -992,7 +992,7 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 		&& !typ_sym.has_method(node.name) {
 		unwrapped_rec_type = (typ_sym.info as ast.Alias).parent_type
 		typ_sym = g.table.sym(unwrapped_rec_type)
-	} else if typ_sym.kind == .array && !typ_sym.has_method(node.name) {
+	} else if typ_sym.kind == .array && !typ_sym.has_method(node.name) && node.name != 'str' {
 		typ := g.table.unaliased_type((typ_sym.info as ast.Array).elem_type)
 		typ_idx := g.table.find_type_idx(g.table.array_name(typ))
 		if typ_idx > 0 {

--- a/vlib/v/tests/generic_array_of_alias_test.v
+++ b/vlib/v/tests/generic_array_of_alias_test.v
@@ -1,0 +1,13 @@
+import datatypes
+
+type Distance = int
+
+struct Foo {
+	field datatypes.LinkedList[Distance]
+}
+
+fn test_generic_array_of_alias() {
+	foo := Foo{}
+	println(foo)
+	assert '${foo.field}' == '[]'
+}

